protocol/dubbo/client.go

```diff
@@ -267,11 +268,25 @@ func (c *Client) call(ct CallType, request *Request, response *Response, callbac
 		return errSessionNotExist
 	}
 	defer func() {
+		failNumber := 0
 		if err == nil {
-			c.pool.put(conn)
-			return
+			for {
+				ok := atomic.CompareAndSwapUint32(&c.pool.pushing, 0, 1)
+				if ok {
+					c.pool.poolQueue.pushHead(conn)
+					c.pool.pushing = 0
+					c.pool.ch <- struct{}{}
+					return
```

Comment:
Can these codes be encapsulated in one method, such as `func (p *gettyRPCClientPool) putGettyRpcClient(conn *gettyRPCClient)`?

Reply:
This is the only place to push conn, so i think the code logic can be placed directly here